src/coreclr/jit/codegenarm.cpp

```diff
@@ -1360,7 +1360,8 @@ void CodeGen::genCodeForStoreInd(GenTreeStoreInd* tree)
             instGen_MemoryBarrier();
         }
 
-        GetEmitter()->emitInsLoadStoreOp(ins_Store(type), emitActualTypeSize(type), data->GetRegNum(), tree);
+        var_types strType = varTypeIsFloating(data->TypeGet()) ? data->TypeGet() : type;
```

Comment:
would  not `ins_StoreFromSrc(data->GetRegNum(), type)` work better?

Reply:
Nice, didn't know about this method. Thank you!

Reply:
@sandreenko Is it clear when to use ins_Store and when to use ins_StoreFromSrc? It seems like ins_StoreFromSrc could (should?) always be used, but currently rarely is.

Reply:
yes, you are right it could and should be used instead. However, for now, many code paths that use `ins_Store` don't have enough info for `ins_StoreFromSrc` (https://github.com/dotnet/runtime/pull/46345#discussion_r549785564)